### PR TITLE
chore: bump osv scanner to 1.9.2

### PR DESCRIPTION
**Pull request by zhaohuabing**
Fixes: https://github.com/google/osv-scanner/issues/1453#issuecomment-2549906182
Related: https://github.com/envoyproxy/gateway/issues/4945

### CODE REVIEW

**Review by zirain — DISMISSED** [review]

**Comment by shahar-h** [comment]
@zhaohuabing you can uncomment [this](https://github.com/envoyproxy/gateway/blob/37d3ce67e209b521a3613fa3463d954dc4882077/tools/osv-scanner/license-scan-config.toml#L5) now.

**Review by zirain — DISMISSED** [review]

File: tools/osv-scanner/license-scan-config.toml
Comment (shahar-h) [inline tools/osv-scanner/license-scan-config.toml:1]: I meant to uncomment `vulnerability.ignore = true` and remove TODO comment.

**Review by zirain — APPROVED** [review]

**Comment by shahar-h** [comment]
@zhaohuabing license scan fails because `github.com/docker/cli@v27.4.0+incompatible` license is not identified. I opened an [issue](https://github.com/google/deps.dev/issues/146) in deps.dev. For now you can add license override in [license-scan-config.toml](tools/osv-scanner/license-scan-config.toml):
```toml
[[PackageOverrides]]
name = "github.com/docker/cli"
version = "v27.4.0+incompatible"
ecosystem = "Go"
license.override = ["Apache-2.0"]
reason = "Unidentified license, remove once https://github.com/google/deps.dev/issues/146 is resolved"
```

**Comment by zirain** [comment]
```
+-------------------+-----------+-------------------------------------------------------------------+-----------------------------------+--------+
| LICENSE VIOLATION | ECOSYSTEM | PACKAGE                                                           | VERSION                           | SOURCE |
+-------------------+-----------+-------------------------------------------------------------------+-----------------------------------+--------+
| UNKNOWN           | Go        | github.com/docker/cli                                             | 27.4.0+incompatible               | go.mod |
| UNKNOWN           | Go        | github.com/docker/docker                                          | 27.4.1+incompatible               | go.mod |
| UNKNOWN           | Go        | go.opentelemetry.io/otel                                          | 1.33.0                            | go.mod |
| UNKNOWN           | Go        | go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc | 1.33.0                            | go.mod |
| UNKNOWN           | Go        | go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp | 1.33.0                            | go.mod |
| UNKNOWN           | Go        | go.opentelemetry.io/otel/exporters/prometheus                     | 0.55.0                            | go.mod |
| UNKNOWN           | Go        | go.opentelemetry.io/otel/exporters/stdout/stdoutmetric            | 1.33.0                            | go.mod |
| UNKNOWN           | Go        | go.opentelemetry.io/otel/metric                                   | 1.33.0                            | go.mod |
| UNKNOWN           | Go        | go.opentelemetry.io/otel/sdk                                      | 1.33.0                            | go.mod |
| UNKNOWN           | Go        | go.opentelemetry.io/otel/sdk/metric                               | 1.33.0                            | go.mod |
| UNKNOWN           | Go        | go.opentelemetry.io/otel/trace                                    | 1.33.0                            | go.mod |
| UNKNOWN           | Go        | google.golang.org/genproto/googleapis/api                         | 0.0.0-202[41](https://github.com/envoyproxy/gateway/actions/runs/12478808774/job/34826939541#step:4:42)209162323-e6fa225c2576 | go.mod |
| UNKNOWN           | Go        | google.golang.org/genproto/googleapis/rpc                         | 0.0.0-20241209162323-e6fa225c2576 | go.mod |
| UNKNOWN           | Go        | helm.sh/helm/v3                                                   | 3.16.4                            | go.mod |
+-------------------+-----------+-------------------------------------------------------------------+-----------------------------------+--------+
```

**Comment by zirain** [comment]
we can not ignore them all one by one.

**Comment by shahar-h** [comment]
> we can not ignore them all one by one.

Wow! I'll open an issue in deps.dev regarding all of these go packages old data and update [here](https://github.com/envoyproxy/gateway/issues/4945). 
For now we can add  `continue-on-error: true` to avoid failing on license violations:
https://github.com/envoyproxy/gateway/blob/e5f86303d83d732f8c3890a6868c62e29a3e3029/.github/workflows/license-scan.yml#L27

File: .github/workflows/license-scan.yml [hunk 1]
```diff
@@ -18,7 +18,8 @@ jobs:
     - name: Checkout code
       uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683  # v4.2.2
     - name: Run scanner
-      uses: google/osv-scanner-action/osv-scanner-action@19ec1116569a47416e11a45848722b1af31a857b  # v1.9.0
+      uses: google/osv-scanner-action/osv-scanner-action@f8115f2f28022984d4e8070d2f0f85abcf6f3458  # v1.9.2
+      continue-on-error: true # remove this after https://github.com/google/deps.dev/issues/146 has been resolved
```
Comment (shahar-h) [inline .github/workflows/license-scan.yml:22]: Add a space before comment

**Review by zirain — APPROVED** [review]

**Review by shahar-h — APPROVED** [review]

**Pull request merged**